### PR TITLE
source_stream_serde returns Result<T> instead of T

### PR DESCRIPTION
**Pull request by zzlk**

### CODE REVIEW

File: hydroflow/examples/echoserver/client.rs [hunk 1]
```diff
@@ -20,6 +20,7 @@ pub(crate) async fn run_client(outbound: UdpSink, inbound: UdpStream, opts: Opts
 
         // Print all messages for debugging purposes
         inbound_chan
+            -> filter_map(bincode::Result::ok)
```
Comment (zzlk) [inline hydroflow/examples/echoserver/client.rs:23]: just regular Result::ok

File: hydroflow_lang/src/graph/ops/source_stream_serde.rs [hunk 2]
```diff
@@ -60,7 +60,7 @@ pub const SOURCE_STREAM_SERDE: OperatorConstraints = OperatorConstraints {
         let write_iterator = quote_spanned! {op_span=>
```
Comment (zzlk) [inline hydroflow_lang/src/graph/ops/source_stream_serde.rs:60]: add docs, add eaxmple for flatmap() away error.

**Comment by MingweiSamuel** [comment]
I'd actually personally default to `Result::unwrap`

**Comment by zzlk** [comment]
> I'd actually personally default to `Result::unwrap`

Oh yeah, this is definitely the right way to do it to preserve the current behavior.

File: hydroflow_lang/src/graph/ops/source_stream_serde.rs [hunk 3]
```diff
@@ -60,7 +60,7 @@ pub const SOURCE_STREAM_SERDE: OperatorConstraints = OperatorConstraints {
...
             let #ident = std::iter::from_fn(|| {
                 match #root::futures::stream::Stream::poll_next(#stream_ident.as_mut(), &mut std::task::Context::from_waker(&#context.waker())) {
-                    std::task::Poll::Ready(Some(std::result::Result::Ok((payload, addr)))) => Some((#root::util::deserialize_from_bytes(payload), addr)),
+                    std::task::Poll::Ready(Some(std::result::Result::Ok((payload, addr)))) => Some(#root::util::deserialize_from_bytes(payload).map(|payload| (payload, addr))),
```
[the remark this answers is not in the document]
Comment (MingweiSamuel) [reply hydroflow_lang/src/graph/ops/source_stream_serde.rs:63]: Add a `pub use bincode;` to lib.rs and its fine

**Review by MingweiSamuel — APPROVED** [review]

**Pull request merged**